### ISSUE TITLE
Trailing newlines end up in buildctl command's build args options

### ISSUE THREAD

**Issue by dbednall** [vito/oci-build-task#32]
:wave: :wave: 
There is a missing chomp/trim somewhere in here, and I am not sure where exactly the right place would be to add it. But here's an example where it's an issue:

use-case for context: 
- I have a job that builds an image and pushes it with a tag value got from the git resource, after a `load_var`. 
- in the next step, a second image is built that uses that tag in its `FROM`, referencing the base image it just built.
  - the tag is set via a `BUILD_ARG_` param in this task. 
  - the build arg defaults to `latest` but can be overridden when building to specify the base image's tag.

the Dockerfile of the second image starts something like this:
```dockerfile
ARG BASE_IMG_TAG=latest
FROM the-base-image:${BASE_IMG_TAG}

# etc
```
and a (pseudoconfig) pipeline like

```yml
resources:
- name: src
  type: git
  icon: git
  source:
    uri: thing.git

- name: foo
  type: registry-image
  icon: docker
  source:
    repository: image-foo

jobs:
- name: build
  plan:
  - get: src
  - load_var: base-img-tag
    file: src/.git/describe_ref # <-------------- !! this guy has a trailing \n !

  # tasks to build and push a base image [...]

  # task to build the next image using the base image:
  - task: build-foo
    privileged: true
    config:
      platform: linux
        image_resource:
          type: registry-image
    	  source: {repository: vito/oci-build-task}
     inputs:
     - name: src
     outputs:
     - name: image
     caches:
     - path: cache
     run:
       path: build
    params: 
      CONTEXT: src
      DOCKERFILE: src/Dockerfile
      BUILD_ARG_BAZ: ((normal-var-works-fine))
      BUILD_ARG_BASE_IMG_TAG: ((.:base-img-tag)) # <-------------- !! this guy!

```

With `debug` param set `true`, you get a log line like
```bash
DEBU[0004] building   buildctl-args="[build --progress plain --frontend dockerfile.v0 --local context=src --local dockerfile=src --opt filename=Dockerfile --output type=docker,dest=/tmp/build/88f95496/image/image.tar --export-cache type=local,mode=min,dest=/tmp/build/88f95496/cache --opt build-arg:BASE_IMG_TAG=4.42.1-86-gfc5e687f\n --opt build-arg:BAZ=no-problem-yo  
```
:x: `--opt build-arg:BASE_IMG_TAG=4.42.1-86-gfc5e687f\n`

which predictably causes
```bash
failed to create LLB definition: failed to parse stage name "the-base-image:4.42.1-86-gfc5e687f\n": invalid reference format
```

---

This works fine by using `.git/short_ref` in the `load_var` step for use as the tag value instead. I guess something to do with how those files are populated makes the difference.
(debug log snippet:
```bash
[...] --opt build-arg:BASE_IMG_TAG=fc5e687 [...]
```
no \n in sight)

So, I'm not sure if this should be fixed 
- in the `git` resource - make the `describe_ref` file be the same as `short_ref` with respect to line endings
- in the `load_vars` step implementation, and (optionally?) trim line endings when loading from `raw` files
- in this task's code, to always trim \n from build args values
- all of the above?

---

guidance appreciated :pray:

**Comment by vito** [comment]
This will be fixed on Concourse v6.5, which defaults `load_var` to a newly added `trim` format. Would you be OK to hold out until then? We expect to ship it next week. :)

**Comment by dbednall** [comment]
Awesome! I’ll close this and sit tight 👍
